mk/tools.mk

```diff
@@ -17,39 +17,37 @@
 ##############################
 
 # Set up ARM (STM32) SDK
-ARM_SDK_BASE_DIR ?= $(TOOLS_DIR)/arm-gnu-toolchain-13.3.rel1
 # Checked below, Should match the output of $(shell arm-none-eabi-gcc -dumpversion)
+# must match arm-none-eabi-gcc-<version> file in arm sdk distribution
 GCC_REQUIRED_VERSION ?= 13.3.1
 
 ## arm_sdk_install   : Install Arm SDK
 .PHONY: arm_sdk_install
 
 # source: https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads
-ifeq ($(OSFAMILY), linux)
+ifeq ($(OSFAMILY)-$(ARCHFAMILY), linux-x86_64)
   ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/13.3.rel1/binrel/arm-gnu-toolchain-13.3.rel1-x86_64-arm-none-eabi.tar.xz
-  ARM_SDK_DIR := $(ARM_SDK_BASE_DIR)-x86_64-arm-none-eabi
-endif
-
-ifeq ($(OSFAMILY), macosx)
-  # Check for Apple Silicon
-  UNAME_PROCESSOR := $(shell uname -p)
-  ifeq ($(UNAME_PROCESSOR), arm)
-    ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/13.3.rel1/binrel/arm-gnu-toolchain-13.3.rel1-darwin-arm64-arm-none-eabi.tar.xz
-    ARM_SDK_DIR := $(ARM_SDK_BASE_DIR)-darwin-arm64-arm-none-eabi
-  else
-    ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/13.3.rel1/binrel/arm-gnu-toolchain-13.3.rel1-darwin-x86_64-arm-none-eabi.tar.xz
-    ARM_SDK_DIR := $(ARM_SDK_BASE_DIR)-darwin-x86_64-arm-none-eabi
-  endif
-endif
-
-ifeq ($(OSFAMILY), windows)
+  DL_CHECKSUM = 0601a9588bc5b9c99ad2b56133b7f118
+else ifeq ($(OSFAMILY)-$(ARCHFAMILY), macosx-x86_64)
+  ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/13.3.rel1/binrel/arm-gnu-toolchain-13.3.rel1-darwin-x86_64-arm-none-eabi.tar.xz
+  DL_CHECKSUM = 4bb141e44b831635fde4e8139d470f1f
+else ifeq ($(OSFAMILY)-$(ARCHFAMILY), macosx-arm64)
+  ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/13.3.rel1/binrel/arm-gnu-toolchain-13.3.rel1-darwin-arm64-arm-none-eabi.tar.xz
+  DL_CHECKSUM = f1c18320bb3121fa89dca11399273f4e
+else ifeq ($(OSFAMILY), windows)
   ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/13.3.rel1/binrel/arm-gnu-toolchain-13.3.rel1-mingw-w64-i686-arm-none-eabi.zip
-  ARM_SDK_DIR := $(ARM_SDK_BASE_DIR)-mingw-w64-i686-arm-none-eabi
+  DL_CHECKSUM = 39d9882ca0eb475e81170ae826c1435d
+else
+  $(error No toolchain URL defined for $(OSFAMILY)-$(ARCHFAMILY))
 endif
 
 ARM_SDK_FILE := $(notdir $(ARM_SDK_URL))
+# remove compression suffixes
+ARM_SDK_DIR := $(TOOLS_DIR)/$(patsubst %.zip,%, 	\
+			    $(patsubst %.tar.xz,%, 	\
+			    $(notdir $(ARM_SDK_URL))))
 
-SDK_INSTALL_MARKER := $(ARM_SDK_DIR)/bin/arm-none-eabi-gcc-$(GCC_REQUIRED_VERSION)
+SDK_INSTALL_MARKER := $(ARM_SDK_DIR)/.installed
```

Comment:
what happens if installation is corrupted somehow and this hidden marker file exists?
test:
delete all contents of ./tools/arm-gnu-toolchain-13.3.rel1-x86_64-arm-none-eabi, but leave `.install`.
`make` fails. `make arm_sdk_install` fails.

but i just don't know if similar corruption could happen in the wild.

Reply:
`git clean -xdf` or `rm-rf downloads` and `rm-rf tools` for the rescue.

Reply:
@nerdCopter : old behavior was to unpack every time (due to wrong dependency handling)..
rm -rf tools is imo good choice,  cleaning old sdk too

Reply:
Prefer not to use `rm -rf ./tools` in any clean recipe for tools (manual is up to user). The reason is we want to keep the old installations from different commits. That way you can go back to a commit prior to a tools update - and still build - without having to reinstall tools again.

Should be `rm -rf $(ARM_SDK_DIR)` for the tool kit in any clean recipe that is valid for this commit.

Reply:
correct. there is no code to `rm`. left to user always.